lib/solargraph/complex_type.rb

```diff
@@ -30,6 +30,16 @@ def qualify api_map, context = ''
       ComplexType.new(types).reduce_object
     end
 
+    # @param generics_to_resolve [Enumerable<String>]]
+    # @param context_type [UniqueType, nil]
+    # @param resolved_generic_values [Hash{String => ComplexType}] Added to as types are encountered or resolved
+    # @return [self]
+    def resolve_generics_from_context generics_to_resolve, context_type, resolved_generic_values: {}
```

Comment:
I'm open to ideas on the naming here - I don't think "context" is the right word.  "from assignment"?  "by pattern matching"?  Not sure.